File: project/compiler/jimmer-compiler-transactional/src/test/resources/transactional/parity/ServiceATx.java
package demo;

import java.lang.Override;
import org.babyfish.jimmer.sql.transaction.Propagation;

@Component
public class ServiceATx extends ServiceA {
    ServiceATx() {
        super();
    }

    @Override
    public int a() {
        return sqlClient.transaction(Propagation.MANDATORY, () ->  {
            return super.a();
        });
    }

    @Override
    protected int b() {
        return sqlClient.transaction(Propagation.REQUIRES_NEW, () ->  {
            return super.b();
        });
    }
}

File: project/compiler/jimmer-compiler/src/test/resources/transactional/parity/ServiceATx.kt
@file:Suppress("warnings")

package demo

import kotlin.Int
import kotlin.Suppress
import org.babyfish.jimmer.sql.transaction.Propagation

@Component
public class ServiceATx() : ServiceA() {
    override fun a(): Int = this.sqlClient.transaction(Propagation.MANDATORY) {
        super.a()
    }

    protected override fun b(): Int = this.sqlClient.transaction(Propagation.REQUIRES_NEW) {
        super.b()
    }
}
